### PR TITLE
Add environment specifications

### LINKED ISSUE #96: spec: document new environment endpoints
# Bug Report

## Describe the bug

Document the new environment endpoints in the OpenAPI spec.

```http
POST /v1/projects/{projectID}/environments
GET /v1/projects/{projectID}/environments
PATCH /v1/environments/{environmentID}
DELETE /v1/environments/{environmentID}
```

## Which versions are affected

Please include component versions and any versioning information that is relevant
to reproduce.

## Reproducer steps

Logical enumerated sequence on how to reproduce the issue.

## Example code or configuration

```
Configuration or code that reproduces the issue
```

## Expected behavior

Please explain what should happen

## Current behavior

Please explain the current behavior

*Disclaimer: Please do not create a Pull Request without creating an issue first.*
Any change needs to be discussed before proceeding.
Failure to do so may result in the rejection of the pull request.

### PR DESCRIPTION
**Pull request by gabrielbussolo**
# Summary of this proposal

Add specifications to the open-api file

## Issue(s) number(s)

Fix #96 

## Checklist for submitter

- [ ] My PR has a related issue/bug number.
- [ ] My PR provides tests
  - [ ] Integration tests are added/passes
  - [ ] Unit tests are added/passes
  - [ ] End-to-end tests added/passes
  - [ ] Static code check added/passes
  - [ ] Linting on documentation added/passes
  - [ ] Does not affect code coverage stats
- [ ] My PR requires updating dependencies
- [ ] My PR has the documentation changes required.

## Checklist for reviewer

- [ ] The proposal fixes a bug/issue or implements a new feature that is well described.
- [ ] The proposal has sufficient test cases that covers the changes.
  - [ ] If changes an API, it does not break backwards compatibility (-1 major version).
- [ ] If integration is required, the proposal has integration tests.
- [ ] The proposal does not break coverage stats
- [ ] The proposal has the required documentation changes.

## Backport

- [ ] Backport to the latest stable release.

### CODE REVIEW

File: spec/open-api.yml [hunk 1]
```diff
@@ -2986,6 +3004,130 @@ paths:
...
+      summary: Create environment
+      operationId: createEnvironment
+      description: Create a environment to a given project.
+      security:
+        - user: [ ]
+        - project: [ ]
+      requestBody:
+        content:
+          application/json:
+            schema:
```
Comment (nicolasparada) [inline spec/open-api.yml:3069]: Can you move the request body schema into `#/components/schemas`?

File: spec/open-api.yml [hunk 2]
```diff
@@ -2986,6 +3004,130 @@ paths:
...
+              type: object
+              properties:
+                name:
+                  type: string
```
Comment (nicolasparada) [inline spec/open-api.yml:3113]:
```suggestion
                  type: string
                  nullable: true
```

File: spec/open-api.yml [hunk 3]
```diff
@@ -2986,6 +3004,130 @@ paths:
...
+      summary: Update environment
+      operationId: updateEnvironments
+      description: Updates a given environment
+      security:
+        - user: [ ]
+        - project: [ ]
+      requestBody:
+        content:
+          application/json:
+            schema:
```
Comment (nicolasparada) [inline spec/open-api.yml:3109]: Can you move the request body schema into `#/components/schemas`?

**Review by niedbalski — APPROVED** [review]
LGTM

**Pull request merged**